python/plugins/processing/algs/qgis/VectorLayerScatterplot3D.py

```diff
@@ -90,6 +116,20 @@ def processAlgorithm(self, parameters, context, feedback):
         yfieldname = self.parameterAsString(parameters, self.YFIELD, context)
         zfieldname = self.parameterAsString(parameters, self.ZFIELD, context)
 
+        title = self.parameterAsString(parameters, self.TITLE, context)
+        xaxis_title = self.parameterAsString(parameters, self.XAXIS_TITLE, context)
+        yaxis_title = self.parameterAsString(parameters, self.YAXIS_TITLE, context)
+        zaxis_title = self.parameterAsString(parameters, self.ZAXIS_TITLE, context)
+
+        if title.strip() == "":
+            title = None
+        if xaxis_title == "":
```

Comment:
Shouldn't this have the same logic for handling " " as the other algorithms?

Reply:
If no title is given, I tell plotly that there is no title. If no x-axis title is given, I use the fieldname as x-axis title. But I still leave the option for no x-axis title at all by passing a single space " "

Reply:
Right, I understand that logic. But you haven't added the same logic to this particular algorithm. Was there a reason why you skipped this one?

Reply:
Ah, I see. The reason is that on 3D plots, plotly automatically adds the labels x y z to the axes if I pass None. But you're right, it still would be better to do the same. I will do this tomorrow.

Reply:
Actually, the current code results in the same behavior for Scatterplot 3D as for the other algorithms: Entering space means there will be no visible legend title. Since the logic on plotlys side is different for 3D charts, the code on the QGIS side must be different to get the same result. On other chart types, passing " " instead of None to plotly would result in an empty axes title that still reserves some white space.

Reply:
Ok, fair enough! It'd be valuable to add a comment to the algorithm outlining this, just in case someone tries to "fix" this in future

Reply:
Sure, I added a comment.